src/ape/managers/chain.py

```diff
@@ -1182,6 +1186,19 @@ def instance_at(
             else:
                 raise  # Current exception
 
+        if abi:
+            try:
```

Comment:
I think instead of the try catch you can do a much smarter approach of supporting different type conversions using successive if statements

So first if it's str, convert to json dict
Then if it's json dict (either as input or from converted string), load as contract type

You can probably add a filename or `pathlib.Path` input support as well

Reply:
All supported types should be listed in the type-declaration for the param.